eth-types/src/lib.rs

```diff
@@ -210,6 +216,23 @@ impl ToWord for usize {
     }
 }
 
+impl ToWord for i32 {
+    fn to_word(&self) -> Word {
+        Word::from(self.unsigned_abs() as u64)
+            * if self.is_negative() {
+                Word::zero() - Word::one()
+            } else {
+                Word::one()
+            }
```

Comment:
```suggestion
        let value = Word::from(self.unsigned_abs() as u64);
        if self.is_negative() {
            value.overflowing_neg().0
        } else {
            value
        }
```

Sorry forgot to test this before and this actually threw an error for negative numbers, new version is actually tested and should work.

Reply:
Updated.